--- ../../bochs_REL_2_6_6_FINAL/vs2008/bochs.vcproj	Mon Jun  9 20:22:50 2014
+++ ../../../../FtpUsr/FtpRoot/vs2008/bochs.vcproj	Sat Aug 16 20:33:24 2014
@@ -69,7 +69,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="winmm.lib comctl32.lib ws2_32.lib iphlpapi.lib"
+				AdditionalDependencies="winmm.lib comctl32.lib ws2_32.lib iphlpapi.lib rpcrt4.lib"
 				OutputFile=".\..\obj-debug\bochs.exe"
 				LinkIncremental="2"
 				SuppressStartupBanner="true"
@@ -163,7 +163,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="winmm.lib comctl32.lib ws2_32.lib iphlpapi.lib"
+				AdditionalDependencies="winmm.lib comctl32.lib ws2_32.lib iphlpapi.lib rpcrt4.lib"
 				OutputFile=".\..\obj-release\bochs.exe"
 				LinkIncremental="1"
 				SuppressStartupBanner="true"
