--- /c/Users/PROGMAN/Downloads/vs2013/bx_debug.vcxproj	Sat Jul 12 08:59:34 2014
+++ bx_debug.vcxproj	Thu Oct  2 17:00:16 2014
@@ -53,6 +53,9 @@
     <BuildLog>
       <Path>$(IntDir)BuildLog-$(TargetName).log</Path>
     </BuildLog>
+    <PreBuildEvent>
+      <Command>midl /client none /out $(SolutionDir)\..\bx_debug\rpc_gen\ $(SolutionDir)\..\bx_debug_rpc\BochsRpcDbg.idl</Command>
+    </PreBuildEvent>
     <ClCompile>
       <Optimization>Disabled</Optimization>
       <AdditionalIncludeDirectories>..;..\instrument\stubs;..\build\win32\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
@@ -83,6 +86,9 @@
     <BuildLog>
       <Path>$(IntDir)BuildLog-$(TargetName).log</Path>
     </BuildLog>
+    <PreBuildEvent>
+      <Command>midl /client none /out $(SolutionDir)\..\bx_debug\rpc_gen\ $(SolutionDir)\..\bx_debug_rpc\BochsRpcDbg.idl</Command>
+    </PreBuildEvent>
     <ClCompile>
       <Optimization>MaxSpeed</Optimization>
       <InlineFunctionExpansion>OnlyExplicitInline</InlineFunctionExpansion>
@@ -125,6 +131,8 @@
       <CompileAs Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">CompileAsCpp</CompileAs>
     </ClCompile>
     <ClCompile Include="..\bx_debug\symbols.cc" />
+    <ClCompile Include="..\bx_debug_rpc\BochsRpcDbgServer.cpp" />
+    <ClCompile Include="..\bx_debug\rpc_gen\BochsRpcDbg_s.c" />
   </ItemGroup>
   <ItemGroup>
     <ClInclude Include="..\bochs.h" />
