--- C:/Users/PROGMAN/AppData/Local/Temp/config.cc-revBASE.svn003.tmp.cc	Sun Jun  1 01:24:20 2014
+++ C:/Users/PROGMAN/workspace/bochs_REL_2_6_6_FINAL/config.cc	Tue Sep 30 19:57:20 2014
@@ -1802,7 +1802,7 @@ static const char *get_builtin_variable(const char
     if (!strcmp(varname, "BXSHARE")) {
 #ifdef WIN32
       wsprintf(keyname, "Software\\Bochs %s", VER_STRING);
-      code = RegOpenKeyEx(HKEY_LOCAL_MACHINE, keyname, 0, KEY_READ, &hkey);
+	  code = RegOpenKeyEx(HKEY_CURRENT_USER, keyname, 0, KEY_READ, &hkey);
       if (code == ERROR_SUCCESS) {
         data[0] = 0;
         size = MAX_PATH;
