--- C:/Users/PROGMAN/AppData/Local/Temp/debug.h-revBASE.svn003.tmp.h	Wed Feb 12 04:51:18 2014
+++ C:/Users/PROGMAN/workspace/bochs_REL_2_6_6_FINAL/bx_debug/debug.h	Tue Sep 30 20:43:58 2014
@@ -31,6 +31,10 @@ Bit32u crc32(const Bit8u *buf, int len);
 
 #if BX_DEBUGGER
 
+#ifdef WIN32
+extern bool bx_dbg_is_rpc_debugging;
+#endif
+
 // some strict C declarations needed by the parser/lexer
 #ifdef __cplusplus
 extern "C" {
