--- C:/Users/PROGMAN/AppData/Local/Temp/main.cc-revBASE.svn003.tmp.cc	Sun Dec 29 16:45:28 2013
+++ C:/Users/PROGMAN/workspace/bochs_REL_2_6_6_FINAL/main.cc	Tue Sep 30 20:37:23 2014
@@ -632,7 +632,12 @@ int bx_init_main(int argc, char *argv[])
       if (++arg >= argc) BX_PANIC(("-dbglog must be followed by a filename"));
       else SIM->get_param_string(BXPN_DEBUGGER_LOG_FILENAME)->set(argv[arg]);
     }
+#ifdef WIN32
+	else if (!strcmp("-rpc", argv[arg])) {
+      bx_dbg_is_rpc_debugging = true;
+	}
 #endif
+#endif
     else if (!strcmp("-f", argv[arg])) {
       if (++arg >= argc) BX_PANIC(("-f must be followed by a filename"));
       else bochsrc_filename = argv[arg];
