Index: main.cc
===================================================================
--- main.cc	(revision 12494)
+++ main.cc	(working copy)
@@ -632,6 +632,11 @@
       if (++arg >= argc) BX_PANIC(("-dbglog must be followed by a filename"));
       else SIM->get_param_string(BXPN_DEBUGGER_LOG_FILENAME)->set(argv[arg]);
     }
+#ifdef WIN32
+	else if (!strcmp("-rpc", argv[arg])) {
+      bx_dbg_is_rpc_debugging = true;
+	}
+#endif
 #endif
     else if (!strcmp("-f", argv[arg])) {
       if (++arg >= argc) BX_PANIC(("-f must be followed by a filename"));
