Index: bx_debug/dbg_main.cc
===================================================================
--- bx_debug/dbg_main.cc	(revision 12606)
+++ bx_debug/dbg_main.cc	(working copy)
@@ -44,6 +44,8 @@
 }
 #endif
 
+#include "../bx_debug_rpc/bx_debug_extra.cpp"
+
 // default CPU in the debugger.  For commands like "dump_cpu" it will
 // use the default instead of always dumping all cpus.
 unsigned dbg_cpu = 0;
@@ -317,6 +319,10 @@
     SIM->refresh_ci();
     SIM->set_display_mode(DISP_MODE_CONFIG);
     SIM->get_param_bool(BXPN_MOUSE_ENABLED)->set(0);
+	if (bx_rpc_debugging){
+		bx_rpc_get_command();
+		continue;
+	}
     bx_get_command();
 reparse:
     if ((*tmp_buf_ptr == '\n') || (*tmp_buf_ptr == 0))
@@ -1203,6 +1209,7 @@
       }
     }
   }
+  bx_print_msr_value();
 }
 
 //
Index: .
===================================================================
--- .	(revision 12606)
+++ .	(working copy)

Property changes on: .
___________________________________________________________________
Modified: svn:ignore
## -18,3 +18,7 ##
 ltdlconf.h
 libtool
 .bochs.out
+obj-debug
+obj-release
+vs2013
+bx_debug_rpc
