--- orig/bx_debug.vcxproj	Sat Jul 12 08:59:34 2014
+++ mod/bx_debug.vcxproj	Sat Jan 31 16:35:55 2015
@@ -44,10 +44,12 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <OutDir>.\..\obj-debug\bx_debug\</OutDir>
     <IntDir>.\..\obj-debug\bx_debug\</IntDir>
+    <IncludePath>..\bx_debug_rpc\rpc_gen\$(Configuration)\;$(IncludePath)</IncludePath>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
     <OutDir>.\..\obj-release\bx_debug\</OutDir>
     <IntDir>.\..\obj-release\bx_debug\</IntDir>
+    <IncludePath>..\bx_debug_rpc\rpc_gen\$(Configuration)\;$(IncludePath)</IncludePath>
   </PropertyGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <BuildLog>
@@ -114,7 +116,10 @@
     </Bscmake>
   </ItemDefinitionGroup>
   <ItemGroup>
-    <ClCompile Include="..\bx_debug\dbg_main.cc" />
+    <ClCompile Include="..\bx_debug\dbg_main.cc">
+      <ForcedIncludeFiles Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">bx_debug_rpc\bx_debug_extra.cpp;%(ForcedIncludeFiles)</ForcedIncludeFiles>
+      <ForcedIncludeFiles Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">bx_debug_rpc\bx_debug_extra.cpp;%(ForcedIncludeFiles)</ForcedIncludeFiles>
+    </ClCompile>
     <ClCompile Include="..\bx_debug\lexer.c">
       <CompileAs Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">CompileAsCpp</CompileAs>
       <CompileAs Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">CompileAsCpp</CompileAs>
@@ -125,6 +130,15 @@
       <CompileAs Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">CompileAsCpp</CompileAs>
     </ClCompile>
     <ClCompile Include="..\bx_debug\symbols.cc" />
+    <ClCompile Include="..\bx_debug_rpc\BochsRpcDbgServer.cpp">
+      <ExceptionHandling Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">Sync</ExceptionHandling>
+      <ExceptionHandling Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Sync</ExceptionHandling>
+    </ClCompile>
+    <ClCompile Include="..\bx_debug_rpc\BochsRpcDbgInterface.cpp">
+      <ExceptionHandling Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">Async</ExceptionHandling>
+      <ExceptionHandling Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Async</ExceptionHandling>
+    </ClCompile>
+    <ClCompile Include="..\bx_debug_rpc\rpc_gen\$(Configuration)\BochsRpcDbg_s.c" />
   </ItemGroup>
   <ItemGroup>
     <ClInclude Include="..\bochs.h" />
@@ -132,6 +146,18 @@
     <ClInclude Include="..\bx_debug\debug.h" />
     <ClInclude Include="..\bx_debug\parser.h" />
     <ClInclude Include="..\bx_debug\syscalls-linux.h" />
+  </ItemGroup>
+  <ItemGroup>
+    <Midl Include="..\bx_debug_rpc\BochsRpcDbg.idl">
+      <DefaultCharType Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">Ascii</DefaultCharType>
+      <DefaultCharType Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Ascii</DefaultCharType>
+      <WarningLevel Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">4</WarningLevel>
+      <WarningLevel Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">4</WarningLevel>
+      <OutputDirectory Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">..\bx_debug_rpc\rpc_gen\$(Configuration)\</OutputDirectory>
+      <OutputDirectory Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">..\bx_debug_rpc\rpc_gen\$(Configuration)\</OutputDirectory>
+      <HeaderFileName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(Filename).h</HeaderFileName>
+      <HeaderFileName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(Filename).h</HeaderFileName>
+    </Midl>
   </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
